### PR TITLE
Add cross join to Python Relational and PySpark API

### PR DESCRIPTION
**Pull request by khalidmammadov**
Adds missing cross join to PySpark and Relational API

### CODE REVIEW

File: src/include/duckdb/common/enums/join_type.hpp [hunk 1]
```diff
@@ -27,10 +27,11 @@ enum class JoinType : uint8_t {
...
 	             // (and NULL if no partner is found)
-	RIGHT_SEMI = 9, // RIGHT SEMI join is created by the optimizer when the children of a semi join need to be switched
-	                // so that the build side can be the smaller table
-	RIGHT_ANTI = 10 // RIGHT ANTI join is created by the optimizer when the children of an anti join need to be
-	                // switched so that the build side can be the smaller table
+	RIGHT_SEMI = 9,  // RIGHT SEMI join is created by the optimizer when the children of a semi join need to be switched
+	                 // so that the build side can be the smaller table
+	RIGHT_ANTI = 10, // RIGHT ANTI join is created by the optimizer when the children of an anti join need to be
+	                 // switched so that the build side can be the smaller table
+	CROSS = 11
```
Comment (Mytherin) [inline src/include/duckdb/common/enums/join_type.hpp:34]: Cross products are not handled as joins in DuckDB. Can we remove this from the internal enum again? We need to create a `CrossProductRelation` to create a cross product.

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! One comment:

File: src/include/duckdb/common/enums/join_type.hpp (hunk 1 above)
Comment (khalidmammadov) [reply src/include/duckdb/common/enums/join_type.hpp:34]: Sure, will remove and come back

**Comment by khalidmammadov** [comment]
@Mytherin I saw you also changed target branch to `feature` but that got many differences with `main`. And I based my branch on main originally. Not sure if you want me to use that as a base. Reverted to main for now

**Comment by khalidmammadov** [comment]
Hey @Mytherin, any comments?

**Comment by Mytherin** [comment]
Thanks for the changes! LGTM

**Pull request merged**